src/pcrlock/pcrlock.c

```diff
@@ -0,0 +1,5208 @@
+                        log_debug("Skipping NO_ACTION event.");
+                        continue;
+                }
+
+                r = event_log_add_record(el, &record);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add record to event log: %m");
+
+                record->pcr = event->pcrIndex;
+                record->source = path;
+                record->firmware_event_type = event->eventType;
+                record->firmware_payload = memdup(payload, payload_size);
+                if (!record->firmware_payload)
+                        return log_oom();
+                record->firmware_payload_size = payload_size;
+
+                const TPMT_HA *ha, *ha_next = NULL;
+                ha = (const TPMT_HA*) ((const uint8_t*) event + offsetof(TCG_PCR_EVENT2, digests.digests));
+                assert(event->digests.count == n_algorithms);
+
+                for (size_t i = 0; i < n_algorithms; i++, ha = ha_next) {
+                        ha_next = (const TPMT_HA*) ((const uint8_t*) ha + offsetof(TPMT_HA, digest) + algorithms[i].digestSize);
+
+                        if (ha->hashAlg != algorithms[i].algorithmId)
+                                return log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "Hash algorithms in even log record don't match log.");
+
+                        if (!tpm2_hash_alg_to_string(algorithms[i].algorithmId))
+                                continue;
+
+                        r = event_log_record_add_bank(
+                                        record,
+                                        algorithms[i].algorithmId,
+                                        &ha->digest,
+                                        algorithms[i].digestSize,
+                                        /* ret= */ NULL);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to add bank to event log record: %m");
+                }
+
+                /* Try to extract a descriptive text */
+                r = event_log_record_extract_firmware_description(record);
+                if (r < 0)
+                        return r;
+
+                r = event_log_add_algorithms_from_record(el, record);
+                if (r < 0)
+                        return r;
+        }
+
+        return 0;
+}
+
+static int event_log_record_parse_json(EventLogRecord *record, JsonVariant *j) {
+        const char *rectype = NULL;
+        JsonVariant *x, *k;
+        uint64_t u;
+        int r;
+
+        assert(record);
+        assert(j);
+
+        if (!json_variant_is_object(j))
+                return log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "record object is not an object.");
+
+        x = json_variant_by_key(j, "pcr");
+        if (!x)
+                return log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "pcr field missing from TPM measurement log file entry.");
```

Comment:
It's strange to capitalize TPM but not pcr.

Reply:
the field is litterally called "pcr" in lowercase. I'll put it in quotes to make clear that one is literal field name.